### DISCUSSION TITLE
"keystone" is not internal or external command, executable program, or batch file.

### DISCUSSION THREAD

**Discussion by dkostiunin** [keystonejs/keystone#7441] [Questions] [answered]
When I run yarn dev in win10 I get this error "keystone" is not internal or external
command, executable program, or batch file.

**Comment by molomby** [comment, marked as answer]
Maybe post the contents of your `package.json`, to help with troubleshooting.

The way this should work is that, when `yarn dev` is run, yarn resolves the command to `keystone dev`, based on the `dev` script in your `package.json`. Then, the `keystone` part of `keystone dev`, is in turn resolved by yarn to the Keystone CLI, based on [the `bin` configured by the core Keystone `package.json`](https://github.com/keystonejs/keystone/blob/main/packages/core/package.json#L26).

From the error, it sounds like the first part is happening but not the second part. Can you confirm the `@keystone-6/core` package is definitely installed (so run `yarn add @keystone-6/core`)? I can't think what else it could be.

Are you creating this project from scratch yourself or using [`create-keystone-app`](https://github.com/keystonejs/create-keystone-app)?

*edit:* See followup answer in https://github.com/keystonejs/keystone/discussions/7441#discussioncomment-2521244

**Reply by dkostiunin** [reply]
> Maybe post the contents of your `package.json`, to help with troubleshooting.


if I install - 

yarn create keystone-app
cd my-app
yarn dev

everything works

But I wanted to try
https://github.com/keystonejs/keystone/tree/main/examples-staging/ecommerce
https://github.com/keystonejs/keystone/tree/main/examples/blog

these examples tell you to run yarn as root - 
i run command line as administrator then yarn dev and i get this error

Now I installed in both projects
yarn add @keystone-6/core and getting new error in case of ecommerce-
 Starting Keystone
Error: Cannot find module 'C:\nodejs\nextjs\keystone\keystone\node_modules\@keystone-6\auth\dist\keystone-6-auth.cjs.js'. Please verify that the package.json has a valid "main" entry

in case the blog -
Generating GraphQL and Prisma schemas
Error: Could not find prisma-fmt binary. Searched in C:\nodejs\nextjs\keystone\keystone\examples\blog\node_modules\@prisma\sdk\prisma-fmt-windows.exe and C:\nodejs\nextjs\keystone\keystone\examples\blog\node_modules\@ prisma\prisma-fmt-windows.exe

Apparently i need to install the necessary packages

**Reply by molomby** [reply]
> these examples tell you to run yarn as root

Ah, I think there's been a misunderstanding. The docs say to run `yarn` _at_ the root, not _as_ root – that is, run `yarn` _in the root directory of the keystone repo_. I don't think any of the steps need to be run as an Administrator (maybe installing node itself? not sure).

You should be doing something like this...

```sh
git clone git@github.com:keystonejs/keystone.git
cd keystone

# Run yarn *in* the root to install all the packages
yarn

# Change directory to the example and run `yarn dev`
cd examples-staging\ecommerce
yarn dev

# Or...
cd examples\blog
yarn dev
```

I don't have a Windows machine handy to test but this works for me on MacOS and your steps should be similar.

The Keystone codebase is what's known as a monorepo – a whole bunch of packages and apps in a single repository – so it works a bit differently than a stand-alone Keystone app you'd create yourself (like when using `yarn create keystone-app`).

**Reply by dkostiunin** [reply]
Many thanks! Everything worked!
It would not be bad to place this instruction in the appropriate examples for new users.

**Reply by molomby** [reply]
Yeah, agreed. I've put this on our list of docs to be improved.
